src/session/session_api.c

```diff
@@ -488,6 +488,57 @@ err:	API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
 /*
+ * __session_rename --
+ *	WT_SESSION->reset method.
+ */
+static int
+__session_reset(WT_SESSION *wt_session, const char *config)
+{
+	WT_CONNECTION_IMPL *conn;
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+
+	conn = (WT_CONNECTION_IMPL *)wt_session->connection;
+	session = (WT_SESSION_IMPL *)wt_session;
+
+	SESSION_API_CALL(session, reset, config, cfg);
+
+	if (F_ISSET(&session->txn, WT_TXN_RUNNING))
+		WT_ERR_MSG(session, EINVAL, "transaction in progress");
```

Comment:
I agree, technically it's a transaction boundary, but so is `WT_CONNECTION.close`, for that matter. I wouldn't document applications can reasonably use `WT_SESSION.reset` to abort transactions, though.

Because it's an application error to call `WT_SESSION.reset` with a running transaction, I doubt if the application will be prepared to handle this error should it occur. Like you say, the application will call `WT_SESSION.reset`, log any errors, and put the handle into a cache.

In other words, if this ever happens, the application has already made a mistake, and if we don't kill the running transaction, it's likely the application has lost track of a transaction, which is worse?